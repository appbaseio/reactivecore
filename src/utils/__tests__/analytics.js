import getFilterString, { filterComponents, rangeComponents, rangeObjectComponents, parseRangeObject, parseFilterValue } from '../analytics';

test('snapshot filter components', () => {
	expect(filterComponents).toMatchSnapshot();
});

test('snapshot range components', () => {
	expect(rangeComponents).toMatchSnapshot();
});

test('snapshot range object components', () => {
	expect(rangeObjectComponents).toMatchSnapshot();
});

test('parseRangeObject should parse the range object', () => {
	const filterKey = 'filter';
	const rangeObject = {
		start: 1,
		end: 3,
	};
	const res = parseRangeObject(filterKey, rangeObject);
	expect(res).toBe('filter=1~3');
});

test('parseFilterValue should parse string values', () => {
	const componentId = 'id1';
	const componentValues = {
		label: 'Filter',
		value: 'value1',
	};
	const res = parseFilterValue(componentId, componentValues);
	expect(res).toBe('Filter=value1');
});

test('parseFilterValue should parse string values and fallback to componentId', () => {
	const componentId = 'id1';
	const componentValues = {
		value: 'value1',
	};
	const res = parseFilterValue(componentId, componentValues);
	expect(res).toBe('id1=value1');
});

test('parseFilterValue should parse array values', () => {
	const componentId = 'id1';
	const componentValues = {
		label: 'Filter',
		value: ['value1', 'value2'],
	};
	const res = parseFilterValue(componentId, componentValues);
	expect(res).toBe('Filter=value1,Filter=value2');
});

test('parseFilterValue should parse array of object values', () => {
	const componentId = 'id1';
	const componentValues = {
		label: 'Filter',
		value: [{ label: 'label1', value: 'value1' }],
	};
	const res = parseFilterValue(componentId, componentValues);
	expect(res).toBe('Filter=value1');
});

test('parseFilterValue should parse array with multiple values', () => {
	const componentId = 'id1';
	const componentValues = {
		label: 'Filter',
		value: [{ label: 'label1', value: 'value1' }, { label: 'label2', value: 'value2' }],
	};
	const res = parseFilterValue(componentId, componentValues);
	expect(res).toBe('Filter=value1,Filter=value2');
});

test('parseFilterValue should parse range values for DateRange', () => {
	const componentId = 'id1';
	const componentValues = {
		label: 'Filter',
		componentType: 'DATERANGE',
		value: ['date1', 'date2'],
	};
	const res = parseFilterValue(componentId, componentValues);
	expect(res).toBe('Filter=date1~date2');
});

test('getFilterString should return null for falsy value', () => {
	const selectedValues = {};
	const res = getFilterString(selectedValues);
	expect(res).toBe(null);
});

test('getFilterString should return filter string', () => {
	const selectedValues = {
		filter1: {
			value: 'value1',
			label: 'Filter1',
			componentType: 'SINGLELIST',
		},
		filter2: {
			value: ['value2', 'value3'],
			componentType: 'MULTILIST',
		},
	};
	const res = getFilterString(selectedValues);
	expect(res).toBe('Filter1=value1,filter2=value2,filter2=value3');
});
