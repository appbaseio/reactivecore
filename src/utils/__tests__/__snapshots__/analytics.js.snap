// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot filter components 1`] = `
Array [
  "NUMBERBOX",
  "TAGCLOUD",
  "TOGGLEBUTTON",
  "DATEPICKER",
  "DATERANGE",
  "MULTIDATALIST",
  "MULTIDROPDOWNLIST",
  "MULTILIST",
  "SINGLEDATALIST",
  "SINGLEDROPDOWNLIST",
  "SINGLELIST",
  "DYNAMICRANGESLIDER",
  "MULTIDROPDOWNRANGE",
  "MULTIRANGE",
  "RANGESLIDER",
  "RATINGSFILTER",
  "SINGLEDROPDOWNRANGE",
  "SINGLERANGE",
]
`;

exports[`snapshot range components 1`] = `
Array [
  "DATERANGE",
  "DYNAMICRANGESLIDER",
  "RANGESLIDER",
  "RANGEINPUT",
  "RATINGSFILTER",
]
`;

exports[`snapshot range object components 1`] = `
Array [
  "SINGLERANGE",
  "SINGLEDROPDOWNRANGE",
  "MULTIRANGE",
  "MULTIDROPDOWNRANGE",
]
`;
